### DISCUSSION TITLE
How do I first-run GitExtensions with WSL git only?

### DISCUSSION THREAD

**Discussion by MicahZoltu** [gitextensions/gitextensions#11022] [Q&A] [answered]
IIUC, GitExtentsions supports using git installed in WSL.  However, when I first launch GitExtensions (Portable) it asks me to tell it where `git.exe` is and there doesn't appear to be any way to bypass this prompt and get to the settings screen where I can enter whatever it is I need to enter to use WSL Git (figuring that out was my next step, but I got blocked at this first step).

**Comment by MicahZoltu** [comment]
I tried putting `\\wsl.localhost\usr\bin\git` into the Find dialog but when I click Open the dialog just closes and GitExtensions exists.

**Reply by gerhardol** [reply, marked as answer]
Windows Git is still required.
For repos in \\wsl$, "wsl git" is used automatically (Git must be in your path).

**Reply by MicahZoltu** [reply]
Would it be reasonable to make a feature request to remove the native `git` requirement?

I am doing all of my development in WSL and Docker containers and trying to keep as much dev tooling off of my Windows host OS as possible.  The exceptions I'm making are for GUI tools like VSCode and GitExtensions.  VSCode works incredibly well with WSL/Docker backend, and GitExtensions is just about the only other GUI tool I find myself using.

**Reply by gerhardol** [reply]
> Would it be reasonable to make a feature request to remove the native git requirement?

Windows native Git is used for startup checks etc, wsl-git is an add-on. Windows Git are used in some situations.
Windows Git can be used as well (if you map wsl to a drive letter), just slower.

I suggest you just install a portable Git distribution without tampering with wslgit and similar.
(wslgit for instance just accesses the default WSL instance and is not adapted to all Git commands, so it is unknown what is not working, GE WSL usage transforms known commands only.)

**Comment by MicahZoltu** [comment]
For others coming along, I added https://github.com/andy-5/wslgit in Windows along with a symlink from `git.exe` to `wslgit.exe` and put both in my PATH.  Then I pointed GitExtensions at it when prompted and things *roughly* seem to be working.  I haven't done much yet (just a fetch to make sure stuff generally worked) but so far no major problems.
